internal/engine/upper_test.go

```diff
@@ -1815,7 +1815,10 @@ func TestDockerComposeRecordsRunLogs(t *testing.T) {
 	f := newTestFixture(t)
 	m, _ := f.setupDCFixture()
 	expected := "hello world"
-	f.setDCRunLogOutput(m.DockerComposeTarget(), expected)
+	output := make(chan string, 1)
+	output <- expected
+	defer close(output)
+	f.setDCRunLogOutput(m.DockerComposeTarget(), output)
```

Comment:
yay so much better thank you! could all this happen in `setDCRunLogOutput` tho? (i.e. initialize the channel if it doesn't exist, and send the output to the channel)

Reply:
ya, i went down that path initially, but it making it thread-safe and closing the channel properly got complex

Reply:
(creating the channel outside the method lets the caller use `defer`)